src/utils/utils.js

```diff
@@ -22,6 +25,14 @@ const waitForElement = (selector, target) => {
       subtree: true,
     });
   });
+
+  const timeoutPromise = new Promise((_, reject) => {
+    setTimeout(() => {
+      reject(new Error('Time out to wait for the element'));
+    }, TIME_OUT);
+  });
+
+  return Promise.race([waitForElement, timeoutPromise]);
```

Comment:
What if you will add the `setTimeout` inside the `waitForElement` promise, without wrapping it with another promise and `race`?

Reply:
Same behavior.